File: backend/local-api-module/src/main/java/org/vyatsu/localApiModule/entity/wallet/Wallet.java
package org.vyatsu.localApiModule.entity.wallet;

import jakarta.persistence.*;
import lombok.*;

import java.util.LinkedHashSet;
import java.util.Set;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Data
@Entity
@Table(name = "wallets")
public class Wallet {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "balance", nullable = false)
    private double balance;

    @OneToMany(mappedBy = "wallet")
    private Set<WalletTransaction> walletTransactions = new LinkedHashSet<>();
}

File: app/src/main/java/com/bytebusters/android/mpsystem/entity/wallet/Wallet.kt
package com.bytebusters.android.mpsystem.entity.wallet

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.OneToMany
import jakarta.persistence.Table
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Data
@Entity
@Table(name = "wallets")
class Wallet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private val id: Long? = null

    @Column(name = "balance", nullable = false)
    private val balance = 0.0

    @OneToMany(mappedBy = "wallet")
    private val walletTransactions: Set<WalletTransaction> = LinkedHashSet()
}
